File: backend/db/seed.py
import pathlib

from backend.src.db.client import get_connection


def run_seeds() -> None:
    seeds_dir = pathlib.Path(__file__).parent / "seeds"
    sql_files = sorted(seeds_dir.glob("*.sql"))

    conn = get_connection()
    try:
        with conn.cursor() as cur:
            cur.execute("SET app.tenant_id = 'everops'")
        for sql_file in sql_files:
            print(f"Seeding {sql_file.name}...")
            sql = sql_file.read_text()
            with conn.cursor() as cur:
                cur.execute(sql)
        conn.commit()
        print("Seeds complete.")
    finally:
        conn.close()


def main() -> None:
    run_seeds()


if __name__ == "__main__":
    main()

File: backend/src/db/seed.ts
import * as fs from 'fs';
import * as path from 'path';
import { pool } from './client';

export async function runSeeds(): Promise<void> {
  const seedsDir = path.join(__dirname, '../../db/seeds');
  const files = fs
    .readdirSync(seedsDir)
    .filter((f) => f.endsWith('.sql'))
    .sort();

  const client = await pool.connect();
  try {
    await client.query("SET app.tenant_id = 'everops'");
    for (const file of files) {
      const sql = fs.readFileSync(path.join(seedsDir, file), 'utf8');
      await client.query(sql);
    }
  } finally {
    client.release();
  }
}
